h2/src/main/org/h2/mvstore/MVStoreTool.java

```diff
@@ -122,6 +122,12 @@ public static void dump(String fileName, Writer writer, boolean details) {
             long pageCount = 0;
             for (long pos = 0; pos < fileSize;) {
                 block.rewind();
+                // Fixbug - An IllegalStateException that wraps EOFException is thrown when partial writes happens 
+                //in the case of power off or file system issues.
+                // So we should first check the read length of block, and skip the broken block at end of DB file.
+                if(pos + block.limit() > fileSize) {
+                    break;
```

Comment:
I think in this case we should be printing out a message e.g. see further down, we do
   pw.printf("ERROR illegal position %d%n", p);

Reply:
@grandinj OK. I'll add such hint message.